graylog2-server/src/main/java/org/graylog2/shared/rest/resources/ProxiedResource.java

```diff
@@ -135,37 +204,80 @@ protected ProxiedResource(@Context HttpHeaders httpHeaders,
                 .findFirst()
                 .orElseThrow(() -> new IllegalStateException("No active master node found"));
 
-        final RemoteInterfaceType remoteInterfaceType = remoteInterfaceProvider.apply(masterNode.getNodeId())
-                .orElseThrow(() -> new IllegalStateException("Master node " + masterNode.getNodeId() + " not found"));
+        return doNodeApiCall(masterNode.getNodeId(), remoteInterfaceProvider, remoteInterfaceFunction, Function.identity());
+    }
+
+    private  <RemoteInterfaceType, RemoteCallResponseType, FinalResponseType> MasterResponse<FinalResponseType> doNodeApiCall(
+            String nodeId,
+            Function<String, Optional<RemoteInterfaceType>> remoteInterfaceProvider,
+            Function<RemoteInterfaceType, Call<RemoteCallResponseType>> remoteInterfaceFunction,
+            Function<RemoteCallResponseType, FinalResponseType> transformer
+    ) throws IOException {
+        final RemoteInterfaceType remoteInterfaceType = remoteInterfaceProvider.apply(nodeId)
+                .orElseThrow(() -> new IllegalStateException("Node " + nodeId + " not found"));
 
         final Call<RemoteCallResponseType> call = remoteInterfaceFunction.apply(remoteInterfaceType);
         final Response<RemoteCallResponseType> response = call.execute();
 
         final byte[] errorBody = response.errorBody() == null ? null : response.errorBody().bytes();
 
-        return MasterResponse.create(response.isSuccessful(), response.code(), response.body(), errorBody);
+        return MasterResponse.create(response.isSuccessful(), response.code(), transformer.apply(response.body()), errorBody);
+    }
+
+    /**
+     * This wrapper is intended to provide additional server error information
+     * if something went wrong beyond the actual API HTTP call.
+     */
+    @AutoValue
+    public static abstract class CallResult<ResponseType> {
+
+        @JsonProperty("success")
```

Comment:
This is bit misleading, because we have two keys called `success` now

```{
    "cbb04ecf-ced3-47a5-9573-4f83b2dce963": {
        "response": {
            "code": 404,
            "entity": null,
            "error_text": "{\"type\":\"ApiError\",\"message\":\"Lookup table <609aa4683373f118e6dc17ed> not found\"}",
            "success": false
        },
        "server_error_message": null,
        "success": true
    }
}
```

maybe we could call this `request_delivered` instead?

Reply:
https://github.com/Graylog2/graylog2-server/pull/10605/commits/e6fdd831b6e5deffc17c2168651afb5614627012